lib/krane/kubernetes_resource.rb

```diff
@@ -497,10 +509,21 @@ def validate_spec_with_kubectl(kubectl)
       end
     end
 
-    def validate_with_dry_run_option(kubectl, dry_run_option)
-      command = ["apply", "-f", file_path, dry_run_option, "--output=name"]
+    # Server side dry run is only supported on apply
+    def validate_with_server_side_dry_run(kubectl)
+      command = ["apply", "-f", file_path, "--server-dry-run", "--output=name"]
+      kubectl.run(*command, log_failure: false, output_is_sensitive: sensitive_template_content?,
+        retry_whitelist: [:client_timeout], attempts: 3)
+    end
+
+    # Local dry run is supported on only create and apply
+    # If the deploy method is create, validating with apply will fail
+    # If the resource template uses generateName, validating with apply will fail
+    def validate_with_local_dry_run(kubectl)
+      verb = deploy_method == :apply ? "apply" : "create"
```

Comment:
From testing, we can't validate using `replace` + `generateName` either. So I switched it up to `create` since that supports `generateName` and works.